jaeger_client/metrics.py

```diff
@@ -21,69 +21,155 @@
 from __future__ import absolute_import
 
 
-class Metrics(object):
-    """Provides an abstraction of metrics reporting framework."""
+class MetricsFactory(object):
+    """Generates new metrics."""
+
+    def __init__(self, count=None, gauge=None, timing=None, tags=None):
+        """
+        :param count: function (key, value) to emit counters
+        :param gauge: function (key, value) to emit gauges
+        :param timing: function (key, value) to emit timings
+        :param tags: {k:v} dictionary
+        """
+        self._count = count
+        self._gauge = gauge
+        self._timing = timing
+        self._tags = tags
+        if not callable(self._count):
+            self._count = None
+        if not callable(self._gauge):
+            self._gauge = None
+        if not callable(self._timing):
+            self._timing = None
```

Comment:
I'm not following this 100%, I defaulted to no-op functions but changing the public method on the base class to return _ doesn't make sense to me given that this means the API will be different for the BaseClass vs the Subclasses